Add initial version of 3scale alert SOPs included on 3scale v2.9 (#9)
3scale/3scale-Operations: sops/alerts/zync_que_scheduled_job_count_high.adoc

:toc:
:toc-placement!:

= ThreescaleZyncQueScheduledJobCountHigh

toc::[]

== Description

* This alert will trigger if the number of `zync-que` **scheduled** jobs is greater than a given threshold
* **Scheduled** jobs are the ones that are enqueued to be executed some time in the future, but not now (never failed, nor got expired)

== Troubleshooting

> * Check if `zync-que` pods might be failing (so no consuming any job from postgres)
> * Check if `zync-que` pods might not be consuming jobs at the required speed (too much cpu usage causing slowliness?)

Author: ```suggestion
* Check if `zync-que` pods might be failing and the jobs are not being consumed at all from zync's postgres
* Check if `zync-que` pods might not be consuming jobs at the required throughput:
```

:toc:
:toc-placement!:

= ThreescaleZyncQueScheduledJobCountHigh

toc::[]

== Description

* This alert will trigger if the number of `zync-que` **scheduled** jobs is greater than a given threshold

> * **Scheduled** jobs are the ones that have been scheduled to be executed in the future

Author: ```suggestion
* **Scheduled** jobs are the ones that are enqueued to be executed some time in the future, but not now (never failed, nor got expired)
```

== Troubleshooting

* Check if `zync-que` pods might be failing (so no consuming any job from postgres)
* Check if `zync-que` pods might not be consuming jobs at the required speed (too much cpu usage causing slowliness?)
- You may need to consider scaling horinzontally `zync-que` deployment
- You may need to consider increasing `zync-que` deployment resources requests/limits

> * Check if `zync` might not be be able to deal with the incoming traffic, generating too many jobs at postgres (maybe `zync` is being attacked?)

Author: ```suggestion
```

> * Check if postgres is having some issues

Author: ```suggestion
* Check if zync's postgres is having issues
```
